Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp

```diff
@@ -390,24 +414,31 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
               // z component on x-faces of grid
               M_xface(i, j, k, 2) = MacroscopicProperties::updateM_field(i, j, k, 2, a_temp_xface, b_temp_static_xface);
 
-              if (normalization_type == 1) {
-
-                  // temporary normalized magnitude of M_xface field at the fixed point
-                  // re-investigate the way we do Ms interp, in case we encounter the case where Ms changes across two adjacent cells that you are doing interp
-                  amrex::Real mag_normalized = std::sqrt( std::pow(M_xface(i, j, k, 0),2.0) + std::pow(M_xface(i, j, k, 1),2.0) +
-                                                          std::pow(M_xface(i, j, k, 2),2.0) )
-                      / MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr);
+              // temporary normalized magnitude of M_xface field at the fixed point
+              // re-investigate the way we do Ms interp, in case we encounter the case where Ms changes across two adjacent cells that you are doing interp
+              amrex::Real M_magnitude_normalized = std::sqrt( std::pow(M_xface(i, j, k, 0),2.0) + std::pow(M_xface(i, j, k, 1),2.0) + 
+                      std::pow(M_xface(i, j, k, 2),2.0) ) / MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr);
 
+              if (M_normalization == 1) {
                   // check the normalized error
-                  if ( amrex::Math::abs(1._rt-mag_normalized) > mag_normalized_error ){
+                  if ( amrex::Math::abs(1._rt-M_magnitude_normalized) > mag_normalized_error ){
                       printf("i = %d, j=%d, k=%d\n", i, j, k);
-                      printf("mag_normalized = %f, mag_normalized_error=%f\n", mag_normalized, mag_normalized_error);
+                      printf("M_magnitude_normalized = %f, mag_normalized_error=%f\n", M_magnitude_normalized, mag_normalized_error);
                       amrex::Abort("Exceed the normalized error of the M_xface field");
                   }
                   // normalize the M_xface field
-                  M_xface(i,j,k,0) /= mag_normalized;
-                  M_xface(i,j,k,1) /= mag_normalized;
-                  M_xface(i,j,k,2) /= mag_normalized;
+                  M_xface(i,j,k,0) /= M_magnitude_normalized;
+                  M_xface(i,j,k,1) /= M_magnitude_normalized;
+                  M_xface(i,j,k,2) /= M_magnitude_normalized;
+              }
+              else if (M_normalization == 0){
```

Comment:
See my logic in the first-order scheme for renormalizing if |Ms| drifts slightly above Ms, as controlled by mag_normalized_error.